UndertaleModLib/UndertaleChunks.cs

```diff
@@ -714,84 +718,145 @@ internal override uint UnserializeObjectCount(UndertaleReader reader)
+                else
+                    nextOffset = reader.ReadInt32(); // (pointer to next element in the layer list)
 
-                        // Actually perform the length checks, depending on layer data
-                        reader.AbsPosition = jumpOffset;
-                        switch ((LayerType)reader.ReadInt32())
-                        {
-                            case LayerType.Background:
-                                if (nextOffset - reader.AbsPosition > 16 * 4)
-                                    reader.undertaleData.SetGMS2Version(2022, 1);
-                                finished = true;
-                                break;
-                            case LayerType.Instances:
-                                reader.Position += 6 * 4;
-                                int instanceCount = reader.ReadInt32();
-                                if (nextOffset - reader.AbsPosition != (instanceCount * 4))
-                                    reader.undertaleData.SetGMS2Version(2022, 1);
-                                finished = true;
-                                break;
-                            case LayerType.Assets:
-                                reader.Position += 6 * 4;
-                                int tileOffset = reader.ReadInt32();
-                                if (tileOffset != reader.AbsPosition + 8)
-                                    reader.undertaleData.SetGMS2Version(2022, 1);
-                                finished = true;
-                                break;
-                            case LayerType.Tiles:
-                                reader.Position += 7 * 4;
-                                int tileMapWidth = reader.ReadInt32();
-                                int tileMapHeight = reader.ReadInt32();
-                                if (nextOffset - reader.AbsPosition != (tileMapWidth * tileMapHeight * 4))
-                                    reader.undertaleData.SetGMS2Version(2022, 1);
-                                finished = true;
-                                break;
-                            case LayerType.Effect:
-                                reader.Position += 7 * 4;
-                                int propertyCount = reader.ReadInt32();
-                                if (nextOffset - reader.AbsPosition != (propertyCount * 3 * 4))
-                                    reader.undertaleData.SetGMS2Version(2022, 1);
-                                finished = true;
-                                break;
-                        }
-                    }
-                }
+                // Actually perform the length checks, depending on layer data
+                reader.AbsPosition = jumpOffset;
 
-                reader.Position = returnTo;
+                LayerType layerType = (LayerType)reader.ReadInt32();
+                // Basically the switch's default, as each successful switch case terminates the loop
```

Comment:
*as well as?

Reply:
No, just "as". I could substitute "because", perhaps.

Reply:
ah in that sense. might be more understandable.